### PR TITLE
Add keypoint-detection task guide

### PR DESCRIPTION
**Pull request by merveenoyan**
I added task guide for the recent keypoint detection task and SuperPoint model. I also attempted to add feature matching with this model but apparently this model doesn't do that so it only demonstrates keypoint detection. Later when SuperGlue is merged I will add feature matching.

### CODE REVIEW

**Comment by merveenoyan** [comment]
IDK why the image is broken, it's here https://huggingface.co/datasets/huggingface/documentation-images/blob/main/bee.jpg

File: docs/source/en/tasks/keypoint_detection.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+
+-->
+
+# Keypoint Detection Task Guide
```
Comment (stevhliu) [inline docs/source/en/tasks/keypoint_detection.md:17]:
```suggestion
# Keypoint Detection
```

File: docs/source/en/tasks/keypoint_detection.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+[[open-in-colab]]
+
+Keypoint Detection is a task used to identify and locate specific points of interest within an image. These keypoints, also known as landmarks, represent meaningful features of objects, such as facial landmarks, or object parts. These models take image input and return following outputs:
```
Comment (stevhliu) [inline docs/source/en/tasks/keypoint_detection.md:21]:
```suggestion
Keypoint detection identifies and locates specific points of interest within an image. These keypoints, also known as landmarks, represent meaningful features of objects, such as facial features or object parts. These models take an image input and return the following outputs: 
```

File: docs/source/en/tasks/keypoint_detection.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+-->
+
+# Keypoint Detection Task Guide
+
+[[open-in-colab]]
+
+Keypoint Detection is a task used to identify and locate specific points of interest within an image. These keypoints, also known as landmarks, represent meaningful features of objects, such as facial landmarks, or object parts. These models take image input and return following outputs: 
+
+- **Keypoints and Scores**: Points of interest and how confidently they are predicted.
```
Comment (stevhliu) [inline docs/source/en/tasks/keypoint_detection.md:23]:
```suggestion
- **Keypoints and Scores**: Points of interest and their confidence scores.
```

File: docs/source/en/tasks/keypoint_detection.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+- **Descriptors**: A representation of the image region surrounding each keypoint, capturing its texture, gradient, orientation and other properties.
+
+In this guide, we will show how to extract keypoints from images.
+
+Let's begin. For this tutorial we will use [SuperPoint](./model_doc/superpoint.md) which is a foundation model for keypoint detection.
+
+We can initialize the model and the processor like following.
```
Comment (stevhliu) [inline docs/source/en/tasks/keypoint_detection.md:30]:
```suggestion
For this tutorial, we will use [SuperPoint](./model_doc/superpoint.md), a foundation model for keypoint detection.
```

File: docs/source/en/tasks/keypoint_detection.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+```python
+from transformers import AutoImageProcessor, SuperPointForKeypointDetection
+processor = AutoImageProcessor.from_pretrained("magic-leap-community/superpoint")
+model = SuperPointForKeypointDetection.from_pretrained("magic-leap-community/superpoint")
+```
+
+Get the images. They look like following.
```
Comment (stevhliu) [inline docs/source/en/tasks/keypoint_detection.md:38]:
```suggestion
Let's test the model on the images below.
```

File: docs/source/en/tasks/keypoint_detection.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+```
+
+We can now process our inputs and infer.
+
+```python
+inputs = processor(images,return_tensors="pt").to(model.device, model.dtype)
+outputs = model(**inputs)
+```
+
+The model output looks like following. The model output has relative keypoints, descriptors, masks and scores for each item in the batch. The mask here is a mask that highlights areas of the image where keypoints are present.
```
Comment (stevhliu) [inline docs/source/en/tasks/keypoint_detection.md:72]:
```suggestion
The model output has relative keypoints, descriptors, masks and scores for each item in the batch. The mask highlights areas of the image where keypoints are present.
```

**Review by stevhliu — APPROVED** [review]
Awesome, thanks!

**Pull request merged**